src/Async/README.md

```diff
@@ -0,0 +1,1369 @@
+a `Rejected`, `swap` takes (2) functions as its arguments. The first function
+is used to map the expected `Rejected` value into a `Resolved`, while the
+second goes from `Resolved` to `Rejected`. If no mapping is required on either,
+then `identity` functions can be used in one or both arguments.
+
+<!-- eslint-disable no-console -->
+<!-- eslint-disable no-sequences -->
+```javascript
+const Async = require('crocks/Async')
+const { Rejected, Resolved } = Async
+
+const compose = require('crocks/helpers/compose')
+const identity = require('crocks/combinators/identity')
+const swap = require('crocks/pointfree/swap')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+// length :: String -> Integer
+const length =
+  x => x.length
+
+// repeat :: String -> Number -> String
+const repeat =
+  char => n => char.repeat(n)
+
+// values :: Async String Number -> Async String Number
+const values =
+  swap(length, repeat('a'))
+
+// valueIso :: Async String Number -> Async String Number
+const valueIso =
+  compose(values, values)
+
+// types :: Async a b -> b a
+const types =
+  swap(identity, identity)
+
+// typeIso :: Async a b -> b a
+const typeIso =
+  compose(types, types)
+
+values(Resolved(5))
+  .fork(log('rej'), log('res'))
+//=> rej: "aaaaa"
+
+values(Rejected('aaaaa'))
+  .fork(log('rej'), log('res'))
+//=> res: 5
+
+valueIso(Resolved(5))
+  .fork(log('rej'), log('res'))
+//=> res: 5
+
+valueIso(Rejected('aaaaa'))
+  .fork(log('rej'), log('res'))
+//=> rej: "aaaaa"
+
+types(Resolved(5))
+  .fork(log('rej'), log('res'))
+//=> rej: 5
+
+types(Rejected('aaaaa'))
+  .fork(log('rej'), log('res'))
+//=> res: "aaaaa"
+
+typeIso(Resolved(5))
+  .fork(log('rej'), log('res'))
+//=> res: 5
+
+typeIso(Rejected('aaaaa'))
+  .fork(log('rej'), log('res'))
+//=> rej: "aaaaa"
+```
+
+#### fork
+
+```haskell
+Async e a ~> ((e -> ()), (a -> ())) -> (() -> ())
+Async e a ~> ((e -> ()), (a -> ()), (() -> ())) -> (() -> ())
+```
+
+The `Async` type is lazy and will not be executed until told to do so
+and `fork` is the primary method used for execution. `fork` implements (2)
+signatures depending on the need for clean up in the event of cancellation, but
+both return a function that can be used for cancellation of a given instance.
+
+The first and more common signature takes (2) functions that will have their
+return values ignored. The first function will be run in the event of the
+`Async` instance settling on `Rejected` and will receive as its single argument
+the value or "cause" of rejection. The second function provided will be executed
+in the case of the instance settling on `Resolved` and will receive as its
+single argument the value the `Async` was resolved with.
+
+The second signature is used when any cleanup needs to be performed after a
+given `Async` is cancelled by having the function returned from `fork` called.
+The first to arguments to the signature are the same as the more common
```

Comment:
Some little typos here (in bold):
The first t**w**o arguments to the signature are the same as the more common
signature described above but takes an addition**al** function that can be used
for "**clean up**" after cancellation.